cli/tools/coverage.rs

```diff
@@ -237,29 +237,48 @@ fn collect_coverages(dir: &PathBuf) -> Result<Vec<Coverage>, AnyError> {
   let entries = fs::read_dir(dir)?;
   for entry in entries {
     let json = fs::read_to_string(entry.unwrap().path())?;
-    let coverage: Coverage = serde_json::from_str(&json)?;
-
-    coverages.push(coverage);
-  }
-
-  // TODO(caspervonb) drain_filter would make this cleaner, its nightly at the moment.
-  if coverages.len() > 1 {
-    coverages.sort_by_key(|k| k.script_coverage.url.clone());
-
-    for i in (1..coverages.len() - 1).rev() {
-      if coverages[i].script_coverage.url
-        == coverages[i - 1].script_coverage.url
-      {
-        let current = coverages.remove(i);
-        let previous = &mut coverages[i - 1];
-
-        for function in current.script_coverage.functions {
-          previous.script_coverage.functions.push(function);
+    let new_coverage: Coverage = serde_json::from_str(&json)?;
+
+    let existing_coverage = coverages
+      .iter_mut()
+      .find(|x| x.script_coverage.url == new_coverage.script_coverage.url);
+
+    if let Some(existing_coverage) = existing_coverage {
+      for new_function in new_coverage.script_coverage.functions {
+        let existing_function = existing_coverage
+          .script_coverage
+          .functions
+          .iter_mut()
+          .find(|x| x.function_name == new_function.function_name);
+
```

Comment:
If existing_function is still None here after matching by name, maybe try to find any function with a matching ranges[0]?